pkgs/by-name/u-/u-config/package.nix

```diff
@@ -0,0 +1,46 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  pkg-config
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "u-config";
+  version = "0.33.1";
+
+  src = fetchFromGitHub {
+    owner = "skeeto";
+    repo = "u-config";
+    rev = "refs/tags/v${finalAttrs.version}";
+    hash = "sha256-r1zcXKLqw/gK+9k3SX7OCBaZhvV2ya5VC9O3h+WdkyY=";
+  };
+
+  makeFlags = [
+    "CROSS=${stdenv.cc.targetPrefix}"
+    "CC=${lib.getExe stdenv.cc}"
+  ];
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildPhase = ''
+    runHook preBuild
+    make pkg-config
+    runHook postBuild
+  '';
```

Comment:
```suggestion
  buildTargets = [ "pkg-config" ];
```

Reply:
Using this, it cause the following compilation error:

```these 2 derivations will be built:
  /nix/store/12ly7q48i9agvpwqav0glzmk6hmplfq1-source.drv
  /nix/store/mmz3pqrq1ggckkiik2m58bg7fs0z1wna-u-config-0.33.1.drv
building '/nix/store/12ly7q48i9agvpwqav0glzmk6hmplfq1-source.drv'...

trying https://github.com/skeeto/u-config/archive/refs/tags/v0.33.1.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 29022    0 29022    0     0  38252      0 --:--:-- --:--:-- --:--:-- 67808
unpacking source archive /build/v0.33.1.tar.gz
building '/nix/store/mmz3pqrq1ggckkiik2m58bg7fs0z1wna-u-config-0.33.1.drv'...
Running phase: unpackPhase
unpacking source archive /nix/store/c9hqb42gpgdm5377fm378rr9nsd23myg-source
source root is source
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
no configure script, doing nothing
Running phase: buildPhase
build flags: SHELL=/nix/store/306znyj77fv49kwnkpxmb0j2znqpa8bj-bash-5.2p26/bin/bash CROSS= CC=/nix/store/9bv7dcvmfcjnmg5mnqwqlq2wxfn8d7yi-gcc-wrapper-13.2.0/bin/gcc
/nix/store/9bv7dcvmfcjnmg5mnqwqlq2wxfn8d7yi-gcc-wrapper-13.2.0/bin/gcc -Os -fno-builtin -fno-asynchronous-unwind-tables -o pkg-config.exe win32_main.c -s -nostdlib -Wl,--gc-sections -lkernel32
In file included from win32_main.c:5:
miniwin32.h:28:16: warning: return type defaults to 'int' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wimplicit-int-Wimplicit-int8;;]
   28 | #define W32(r) __declspec(dllimport) r __stdcall
      |                ^~~~~~~~~~
miniwin32.h:29:1: note: in expansion of macro 'W32'
   29 | W32(b32)    CloseHandle(uptr);
      | ^~~
miniwin32.h: In function '__declspec':
miniwin32.h:29:13: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'CloseHandle'
   29 | W32(b32)    CloseHandle(uptr);
      |             ^~~~~~~~~~~
miniwin32.h:28:16: error: expected declaration specifiers before '__declspec'
   28 | #define W32(r) __declspec(dllimport) r __stdcall
      |                ^~~~~~~~~~
miniwin32.h:30:1: note: in expansion of macro 'W32'
   30 | W32(i32)    CreateFileW(c16 *, i32, i32, uptr, i32, i32, i32);
      | ^~~
miniwin32.h:28:16: error: expected declaration specifiers before '__declspec'
   28 | #define W32(r) __declspec(dllimport) r __stdcall
      |                ^~~~~~~~~~
miniwin32.h:31:1: note: in expansion of macro 'W32'
   31 | W32(void)   ExitProcess(i32);
      | ^~~
miniwin32.h:28:16: error: expected declaration specifiers before '__declspec'
   28 | #define W32(r) __declspec(dllimport) r __stdcall
      |                ^~~~~~~~~~
miniwin32.h:32:1: note: in expansion of macro 'W32'
   32 | W32(c16 *)  GetCommandLineW(void);
      | ^~~
miniwin32.h:28:16: error: expected declaration specifiers before '__declspec'
   28 | #define W32(r) __declspec(dllimport) r __stdcall
      |                ^~~~~~~~~~
miniwin32.h:33:1: note: in expansion of macro 'W32'
   33 | W32(b32)    GetConsoleMode(uptr, i32 *);
      | ^~~
miniwin32.h:28:16: error: expected declaration specifiers before '__declspec'
   28 | #define W32(r) __declspec(dllimport) r __stdcall
      |                ^~~~~~~~~~
miniwin32.h:34:1: note: in expansion of macro 'W32'
   34 | W32(i32)    GetEnvironmentVariableW(c16 *, c16 *, i32);
      | ^~~
miniwin32.h:28:16: error: expected declaration specifiers before '__declspec'
   28 | #define W32(r) __declspec(dllimport) r __stdcall
      |                ^~~~~~~~~~
miniwin32.h:35:1: note: in expansion of macro 'W32'
   35 | W32(i32)    GetModuleFileNameW(uptr, c16 *, i32);
      | ^~~
miniwin32.h:28:16: error: expected declaration specifiers before '__declspec'
   28 | #define W32(r) __declspec(dllimport) r __stdcall
      |                ^~~~~~~~~~
miniwin32.h:36:1: note: in expansion of macro 'W32'
   36 | W32(i32)    GetStdHandle(i32);
      | ^~~
miniwin32.h:28:16: error: expected declaration specifiers before '__declspec'
   28 | #define W32(r) __declspec(dllimport) r __stdcall
      |                ^~~~~~~~~~
miniwin32.h:37:1: note: in expansion of macro 'W32'
   37 | W32(b32)    ReadFile(uptr, u8 *, i32, i32 *, uptr);
      | ^~~
miniwin32.h:28:16: error: expected declaration specifiers before '__declspec'
   28 | #define W32(r) __declspec(dllimport) r __stdcall
      |                ^~~~~~~~~~
miniwin32.h:38:1: note: in expansion of macro 'W32'
   38 | W32(byte *) VirtualAlloc(uptr, size, i32, i32);
      | ^~~
miniwin32.h:28:16: error: expected declaration specifiers before '__declspec'
   28 | #define W32(r) __declspec(dllimport) r __stdcall
      |                ^~~~~~~~~~
miniwin32.h:39:1: note: in expansion of macro 'W32'
   39 | W32(b32)    WriteConsoleW(uptr, c16 *, i32, i32 *, uptr);
      | ^~~
miniwin32.h:28:16: error: expected declaration specifiers before '__declspec'
   28 | #define W32(r) __declspec(dllimport) r __stdcall
      |                ^~~~~~~~~~
miniwin32.h:40:1: note: in expansion of macro 'W32'
   40 | W32(b32)    WriteFile(uptr, u8 *, i32, i32 *, uptr);
      | ^~~
In file included from win32_main.c:6:
cmdline.c:21:1: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token
   21 | {
      | ^
win32_main.c:17:3: error: storage class specified for parameter 'handles'
   17 | } handles[3];
      |   ^~~~~~~
win32_main.c:22:3: error: storage class specified for parameter 's16'
   22 | } s16;
      |   ^~~
win32_main.c:24:12: error: expected '=', ',', ';', 'asm' or '__attribute__' before 's16cuthead_'
   24 | static s16 s16cuthead_(s16 s, size off)
      |            ^~~~~~~~~~~
win32_main.c:34:1: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token
   34 | {
      | ^
win32_main.c:45:13: error: storage class specified for parameter 'char32_t'
   45 | typedef i32 char32_t;
      |             ^~~~~~~~
win32_main.c:46:18: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'c32'
   46 | typedef char32_t c32;
      |                  ^~~
win32_main.c:48:1: warning: empty declaration
   48 | enum {
      | ^~~~
win32_main.c:54:5: error: unknown type name 'c32'
   54 |     c32 rune;
      |     ^~~
win32_main.c:55:3: error: storage class specified for parameter 'utf8'
   55 | } utf8;
      |   ^~~~
win32_main.c:57:13: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'utf8decode_'
   57 | static utf8 utf8decode_(s8 s)
      |             ^~~~~~~~~~~
win32_main.c:103:31: error: unknown type name 'c32'; did you mean 'u32'?
  103 | static i32 utf8encode_(u8 *s, c32 rune)
      |                               ^~~
      |                               u32
win32_main.c:124:5: error: expected specifier-qualifier-list before 's16'
  124 |     s16 tail;
      |     ^~~
win32_main.c:126:3: error: storage class specified for parameter 'utf16'
  126 | } utf16;
      |   ^~~~~
win32_main.c:128:14: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'utf16decode_'
  128 | static utf16 utf16decode_(s16 s)
      |              ^~~~~~~~~~~~
win32_main.c:158:35: error: unknown type name 'c32'; did you mean 'u32'?
  158 | static i32 utf16encode_(c16 *dst, c32 rune)
      |                                   ^~~
      |                                   u32
win32_main.c:173:12: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'towide_'
  173 | static s16 towide_(arena *perm, s8 s)
      |            ^~~~~~~
win32_main.c:194:34: error: expected declaration specifiers or '...' before 's16'
  194 | static s8 fromwide_(arena *perm, s16 w)
      |                                  ^~~
win32_main.c:216:1: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token
  216 | {
      | ^
win32_main.c:244:1: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token
  244 | {
      | ^
win32_main.c:250:1: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token
  250 | {
      | ^
win32_main.c:280:1: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token
  280 | {
      | ^
win32_main.c:287:1: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token
  287 | {
      | ^
win32_main.c:300:1: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token
  300 | {
      | ^
win32_main.c:310:1: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token
  310 | {
      | ^
win32_main.c:317:1: error: expected declaration specifiers before '__attribute'
  317 | __attribute((force_align_arg_pointer))
      | ^~~~~~~~~~~
win32_main.c:359:1: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token
  359 | {
      | ^
win32_main.c:408:1: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token
  408 | {
      | ^
win32_main.c:418:3: error: storage class specified for parameter 'u16buf'
  418 | } u16buf;
      |   ^~~~~~
win32_main.c:420:27: error: expected declaration specifiers or '...' before 'u16buf'
  420 | static void flushconsole_(u16buf *b)
      |                           ^~~~~~
win32_main.c:429:23: error: expected declaration specifiers or '...' before 'u16buf'
  429 | static void printc32_(u16buf *b, c32 rune)
      |                       ^~~~~~
win32_main.c:429:34: error: unknown type name 'c32'; did you mean 'u32'?
  429 | static void printc32_(u16buf *b, c32 rune)
      |                                  ^~~
      |                                  u32
win32_main.c:438:1: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token
  438 | {
      | ^
miniwin32.h:28:16: warning: type of 'dllimport' defaults to 'int' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wimplicit-int-Wimplicit-int8;;]
   28 | #define W32(r) __declspec(dllimport) r __stdcall
      |                ^~~~~~~~~~
miniwin32.h:29:1: note: in expansion of macro 'W32'
   29 | W32(b32)    CloseHandle(uptr);
      | ^~~
win32_main.c:418:3: error: declaration for parameter 'u16buf' but no such parameter
  418 | } u16buf;
      |   ^~~~~~
win32_main.c:126:3: error: declaration for parameter 'utf16' but no such parameter
  126 | } utf16;
      |   ^~~~~
win32_main.c:55:3: error: declaration for parameter 'utf8' but no such parameter
   55 | } utf8;
      |   ^~~~
win32_main.c:45:13: error: declaration for parameter 'char32_t' but no such parameter
   45 | typedef i32 char32_t;
      |             ^~~~~~~~
win32_main.c:22:3: error: declaration for parameter 's16' but no such parameter
   22 | } s16;
      |   ^~~
win32_main.c:17:3: error: declaration for parameter 'handles' but no such parameter
   17 | } handles[3];
      |   ^~~~~~~
win32_main.c:478: error: expected '{' at end of input
In file included from win32_main.c:4:
u-config.c: At top level:
u-config.c:61:16: warning: 'os_mapfile' used but never defined
   61 | static filemap os_mapfile(arena *, s8 path);
      |                ^~~~~~~~~~
u-config.c:65:13: warning: 'os_write' used but never defined
   65 | static void os_write(i32 fd, s8);
      |             ^~~~~~~~
u-config.c:68:13: warning: 'os_fail' used but never defined
   68 | static void os_fail(void) __attribute((noreturn));
      |             ^~~~~~~
make: *** [Makefile:14: pkg-config.exe] Error 1
error: builder for '/nix/store/mmz3pqrq1ggckkiik2m58bg7fs0z1wna-u-config-0.33.1.drv' failed with exit code 2;
       last 10 log lines:
       > u-config.c:61:16: warning: 'os_mapfile' used but never defined
       >    61 | static filemap os_mapfile(arena *, s8 path);
       >       |                ^~~~~~~~~~
       > u-config.c:65:13: warning: 'os_write' used but never defined
       >    65 | static void os_write(i32 fd, s8);
       >       |             ^~~~~~~~
       > u-config.c:68:13: warning: 'os_fail' used but never defined
       >    68 | static void os_fail(void) __attribute((noreturn));
       >       |             ^~~~~~~
       > make: *** [Makefile:14: pkg-config.exe] Error 1
       For full logs, run 'nix log /nix/store/mmz3pqrq1ggckkiik2m58bg7fs0z1wna-u-config-0.33.1.drv'.
```

Reply:
Oh yes, try `buildFlags` instead.